Music Player - Step 58 Use the OR operator unclear

### Describe the Issue

The current instructions are a little unclear
```
create a song constant and set it to the current song of userData, or the first song in the userData?.songs array.
```
It's not too clear if you are to set `song` to one of those, or if you are to use the `OR` operator in the assignment. I think there could be an instruction or clue to make this more clear:

```
create a song constant and use the OR operator ( || ) to set it to the current song of userData, or the first song in the userData?.songs array.
```


### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-string-and-array-methods-by-building-a-music-player/step-58

### Your code

```
const setPlayButtonAccessibleText = () => {

};
```


### Expected behavior

 I think there could be an instruction or clue to make this more clear:

```
create a song constant and use the OR operator ( || ) to set it to the current song of userData, or the first song in the userData?.songs array.

```

Additionally the error feedback and the console feedback is different. The feedback in the console is much more explicit. (See screenshot)

### Screenshots

![Screenshot 2023-12-25 104053](https://github.com/freeCodeCamp/freeCodeCamp/assets/79821659/8cc4bb17-4eb1-429f-9e2e-62e284b3d68e)


### System

- Device: e.g. iPhone 6, Laptop
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]


### Additional context

_No response_

@huyenltnguyen 
I followed all the instructions to run this project locally, but when try to run the project with pnpm run develop command, I got this error.

![freecodecamp](https://github.com/freeCodeCamp/freeCodeCamp/assets/29702609/02ce66a2-9b21-4f85-86f7-8f8c6761e444)

 form-group.tsx
```
import React, { createContext } from 'react';
import { FormGroupProps } from './types';

export type FormContextProps = Pick<
  FormGroupProps,
  'controlId' | 'validationState'
>;
export const FormContext = createContext<FormContextProps>({});

const defaultClasses = 'mb-3.5 relative';

export const FormGroup = ({
  className,
  validationState,
  controlId,
  as,
  ...props
}: FormGroupProps): JSX.Element => {
  const context = {
    controlId,
    validationState
  };
  const componentClass = as;
  const Component = componentClass || 'div';

  const classes = [defaultClasses, className].join(' ');
  return (
    <FormContext.Provider value={context}>
      <Component
        className={classes}
        as={as}
        id={controlId}
        {...props}
        validationstate={validationState}
      />
    </FormContext.Provider>
  );
};
```

```
../../../../../node_modules/@types/react/index.d.ts:3436:13 - error TS2717: Subsequent property declarations 
same type.  Property 'rect' must be of type 'SVGProps<SVGRectElement>', but here has type 'SVGProps<SVGRectEl

3436             rect: React.SVGProps<SVGRectElement>;
                 ~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3322:13
    3322             rect: React.SVGProps<SVGRectElement>;
                     ~~~~
    'rect' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3437:13 - error TS2717: Subsequent property declarations 
same type.  Property 'stop' must be of type 'SVGProps<SVGStopElement>', but here has type 'SVGProps<SVGStopEl

3437             stop: React.SVGProps<SVGStopElement>;
                 ~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3323:13
    3323             stop: React.SVGProps<SVGStopElement>;
                     ~~~~
    'stop' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3438:13 - error TS2717: Subsequent property declarations 
same type.  Property 'switch' must be of type 'SVGProps<SVGSwitchElement>', but here has type 'SVGProps<SVGSw
3438             switch: React.SVGProps<SVGSwitchElement>;
                 ~~~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3324:13
    3324             switch: React.SVGProps<SVGSwitchElement>;
                     ~~~~~~
    'switch' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3439:13 - error TS2717: Subsequent property declarations 
same type.  Property 'symbol' must be of type 'SVGProps<SVGSymbolElement>', but here has type 'SVGProps<SVGSy
3439             symbol: React.SVGProps<SVGSymbolElement>;
                 ~~~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3325:13
    3325             symbol: React.SVGProps<SVGSymbolElement>;
                     ~~~~~~
    'symbol' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3440:13 - error TS2717: Subsequent property declarations 
same type.  Property 'text' must be of type 'SVGTextElementAttributes<SVGTextElement>', but here has type 'SVtributes<SVGTextElement>'.

3440             text: React.SVGTextElementAttributes<SVGTextElement>;
                 ~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3326:13
    3326             text: React.SVGTextElementAttributes<SVGTextElement>;
                     ~~~~
    'text' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3441:13 - error TS2717: Subsequent property declarations 
same type.  Property 'textPath' must be of type 'SVGProps<SVGTextPathElement>', but here has type 'SVGProps<Sent>'.

3441             textPath: React.SVGProps<SVGTextPathElement>;
                 ~~~~~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3327:13
    3327             textPath: React.SVGProps<SVGTextPathElement>;
                     ~~~~~~~~
    'textPath' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3442:13 - error TS2717: Subsequent property declarations 
same type.  Property 'tspan' must be of type 'SVGProps<SVGTSpanElement>', but here has type 'SVGProps<SVGTSpa

3442             tspan: React.SVGProps<SVGTSpanElement>;
                 ~~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3328:13
    3328             tspan: React.SVGProps<SVGTSpanElement>;
                     ~~~~~
    'tspan' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3443:13 - error TS2717: Subsequent property declarations 
same type.  Property 'use' must be of type 'SVGProps<SVGUseElement>', but here has type 'SVGProps<SVGUseEleme

3443             use: React.SVGProps<SVGUseElement>;
                 ~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3329:13
    3329             use: React.SVGProps<SVGUseElement>;
                     ~~~
    'use' was also declared here.

../../../../../node_modules/@types/react/index.d.ts:3444:13 - error TS2717: Subsequent property declarations 
same type.  Property 'view' must be of type 'SVGProps<SVGViewElement>', but here has type 'SVGProps<SVGViewEl

3444             view: React.SVGProps<SVGViewElement>;
                 ~~~~

  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3330:13
    3330             view: React.SVGProps<SVGViewElement>;
                     ~~~~
    'view' was also declared here.


Found 182 errors in 3 files.

Errors  Files
     5  ../../node_modules/.pnpm/@types+react@16.14.51/node_modules/@types/react/index.d.ts:3126
     1  src/form-group/form-group.tsx:29
   176  ../../../../../node_modules/@types/react/index.d.ts:3254
/mnt/c/Users/Ashish Gupta/Desktop/open-source/freeCodeCamp/tools/ui-components:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @freecodecamp/ui@0.0.1 build: `pnpm run build:css && pnpm run build:js &&
Exit status 2
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed.
 ELIFECYCLE  Command failed.
ERROR: "develop:client" exited with 1.
 ELIFECYCLE  Command failed with exit code 1.
 ```
 

--- a/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6555dd138e70cae6b546966d.md
+++ b/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6555dd138e70cae6b546966d.md
@@ -7,7 +7,7 @@ dashedName: step-58
 
 # --description--
 
-You need to get the currently playing song or the first song in the playlist. To do that, create a `song` constant and set it to the current song of `userData`, **or** the first song in the `userData?.songs` array.
+You need to get the currently playing song or the first song in the playlist. To do that, create a `song` constant and use the OR operator ( || ) to set it to the current song of `userData`, **or** the first song in the `userData?.songs` array.
 
 Don't forget to use optional chaining.
 

--- a/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6555dd138e70cae6b546966d.md
+++ b/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/6555dd138e70cae6b546966d.md
@@ -7,7 +7,7 @@ dashedName: step-58
 
 # --description--
 
-You need to get the currently playing song or the first song in the playlist. To do that, create a `song` constant and use the OR operator ( || ) to set it to the current song of `userData`, **or** the first song in the `userData?.songs` array.
+You need to get the currently playing song or the first song in the playlist. To do that, create a `song` constant and use the OR operator (`||`) to set it to the current song of `userData`, **or** the first song in the `userData?.songs` array.
 
 Don't forget to use optional chaining.
 

